### PR TITLE
Fix for #941

### PR DESCRIPTION
**Pull request by vzts**
This PR fixes https://github.com/LonamiWebs/Telethon/issues/941
Not sure if it's the best fix. Please review.
Thanks for the awesome library btw!

### CODE REVIEW

File: telethon/client/uploads.py [hunk 1]
```diff
@@ -393,18 +393,19 @@ class UploadMethods(ButtonMethods, MessageParseMethods, UserMethods):
...
             elif not force_document and utils.is_gif(file):
                 media = types.InputMediaGifExternal(file, '')
             else:
                 media = types.InputMediaDocumentExternal(file)
+        elif isinstance(file, str):
+            file_handle = await self.upload_file(
+                file, progress_callback=progress_callback,
+                use_cache=use_cache if allow_cache else None
+            )
         else:
```
Comment (Lonami) [inline telethon/client/uploads.py]: This expect an string so it will never be handled.

File: telethon/client/uploads.py [hunk 2]
```diff
@@ -393,18 +393,19 @@ class UploadMethods(ButtonMethods, MessageParseMethods, UserMethods):
...
         as_image = utils.is_image(file) and not force_document
         use_cache = types.InputPhoto if as_image else types.InputDocument
-        if not isinstance(file, str):
-            file_handle = await self.upload_file(
-                file, progress_callback=progress_callback,
-                use_cache=use_cache if allow_cache else None
-            )
-        elif re.match('https?://', file):
+
+        if re.match('https?://', file):
```
Comment (Lonami) [inline telethon/client/uploads.py]: I'm fairly confident this will raise with non-`str`.

**Comment by Lonami** [comment]
I left a comment in #941 before checking this.

File: telethon/client/uploads.py (hunk 2 above)
Comment (vzts) [reply telethon/client/uploads.py]: You are right. Why did I think it wouldn't raise..? Maybe I was too sleepy last night. I'll fix it right away.

**Comment by vzts** [comment]
Think I understood where it went wrong. Not 100% sure if I'm doing right though. Please review.

**Comment by Lonami** [comment]
This fix is equally valid and while https://github.com/LonamiWebs/Telethon/commit/9195bccb41695a8f47218bc80b0458af821eafbd makes an OS call to check if the file exists, I believe it's OK to prioritize system files over other resources. Sorry, I committed it before checking your update, but appreciate the effort.

**Pull request closed without merging**